services-custom/s3-transfer-manager/src/main/java/software/amazon/awssdk/transfer/s3/internal/progress/TransferProgressUpdater.java

```diff
@@ -136,41 +136,31 @@ public void subscriberOnComplete() {
         };
     }
 
-    public <ResultT> AsyncResponseTransformer<GetObjectResponse, ResultT> wrapResponseTransformer(
+    public <ResultT> AsyncResponseTransformer<GetObjectResponse, ResultT> wrapResponseTransformerForMultipartDownload(
         AsyncResponseTransformer<GetObjectResponse, ResultT> responseTransformer) {
         return AsyncResponseTransformerListener.wrap(
             responseTransformer,
-            new AsyncResponseTransformerListener<GetObjectResponse>() {
+            new BaseAsyncResponseTransformerListener() {
                 @Override
                 public void transformerOnResponse(GetObjectResponse response) {
-                    if (response.contentLength() != null) {
-                            progress.updateAndGet(b -> b.totalBytes(response.contentLength()).sdkResponse(response));
-                    }
-                }
-
-                @Override
-                public void transformerExceptionOccurred(Throwable t) {
-                    transferFailed(t);
-                }
-
-                @Override
-                public void publisherSubscribe(Subscriber<? super ByteBuffer> subscriber) {
-                    resetBytesTransferred();
-                }
-
-                @Override
-                public void subscriberOnNext(ByteBuffer byteBuffer) {
-                    incrementBytesTransferred(byteBuffer.limit());
-                }
-
-                @Override
-                public void subscriberOnError(Throwable t) {
-                    transferFailed(t);
+                    ContentRangeParser contentRangeParser = new ContentRangeParser();
```

Comment:
Is there any reason it's not a util class?

Reply:
No specific reasons beside trying to limit the number of "utils" classes. Since this class has one specific and well bounded use case, I didn't see the need of a util class. But I don't have a string feeling about it.

Reply:
I'm normally not a pro-util person, but this seems like a classic use of util class (parsing and string handling)

Reply:
Plus, micro optimization to avoid object creation/GC